sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1182,6 +1234,79 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
       (c, evPrim, evNull) => code"$evPrim = $c != 0;"
   }
 
+  private[this] def castTimestampToIntegerCode(
+      ctx: CodegenContext,
+      intType: String): CastFunction = {
+    if (failOnIntegerOverflow) {
+      val longValue = ctx.freshName("longValue")
+      (c, evPrim, evNull) =>
+        code"""
+          long $longValue = ${timestampToIntegerCode(c)};
+          if ($longValue == ($intType) $longValue) {
+            $evPrim = ($intType) $longValue;
+          } else {
+            throw new ArithmeticException("Casting $c to ${intType.capitalize} causes overflow");
+          }
+        """
+    } else {
+      (c, evPrim, evNull) => code"$evPrim = ($intType) ${timestampToIntegerCode(c)};"
+    }
+  }
+
+  private[this] def lowerAndUpperBound(intType: String): (Double, Double) = {
+    intType.toLowerCase(Locale.ROOT) match {
+      case "int" => (Int.MinValue - 1.0, Int.MaxValue.toLong + 1.0)
+      case "short" => (Short.MinValue - 1.0, Short.MaxValue + 1.0)
+      case "byte" => (Byte.MinValue - 1.0, Byte.MaxValue + 1.0)
+    }
+  }
+
+  private[this] def castDecimalToIntegerCode(
+      ctx: CodegenContext,
+      intType: String): CastFunction = {
```

Comment:
`intType`? Do you mean `inType`?

Reply:
It's actually `integerType`. But the full name makes some code longer than 100 characters. 
I can change it if you think it is misleading.